sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeTestUtils.scala

```diff
@@ -17,12 +17,16 @@
 
 package org.apache.spark.sql.catalyst.util
 
+import java.time.{LocalDate, LocalDateTime, LocalTime}
 import java.util.TimeZone
+import java.util.concurrent.TimeUnit
+
+import org.apache.spark.sql.catalyst.util.DateTimeUtils.TimeZoneUTC
 
 /**
  * Helper functions for testing date and time functionality.
  */
-object DateTimeTestUtils {
+trait DateTimeTestUtils {
```

Comment:
Does it need to become a trait? not against it but I didn't see why it was needed from skimming this. Is it just convenience?

Reply:
I also have the same concern. If there is no clear benefit, I prefer to keep the previous `object DateTimeTestUtils`.

Also, @MaxGekk . This PR is mostly a refactoring instead of simple `Get rid of java.util.Calendar`. It would be better if we use a matched PR title next time.

Reply:
> If there is no clear benefit, I prefer to keep the previous object DateTimeTestUtils ...

This is mostly for consistency. As I can see we usually mix common code from traits: `ExpressionEvalHelper`, `TestJsonData` and etc. 

> Also, @MaxGekk . This PR is mostly a refactoring instead of simple Get rid of java.util.Calendar. It would be better if we use a matched PR title next time.

The PR changes implementation of 2 methods `truncTimestamp` and `getOffsetFromLocalMillis`, and changes their behavior for dates before 1582 (I will update migration guide). I cannot fully agree it is refactoring. And PR's title reflect actual goal - remove last 2 places (except tests) where the hybrid calendar is used explicitly.

Reply:
Let's get rid of this change `object` to `trait` anyway if that's not related with getting rid of `java.util.Calendar`.

Reply:
I reverted it back